en/components/general/wpf_to_angular_guide/layout.md

```diff
@@ -0,0 +1,245 @@
+
+### StackPanel
+
+The [`StackPanel`](https://docs.microsoft.com/en-us/dotnet/api/system.windows.controls.stackpanel?view=netframework-4.8) arranges its child elements into a single line that can be oriented horizontally or vertically. Let us add several buttons in a StackPanel and see how they look like in WPF:
+
+```xml
+<StackPanel>
+    <Button>1</Button>
+    <Button>2</Button>
+    <Button>3</Button>
+    <Button>4</Button>
+    <Button>5</Button>
+</StackPanel>
+```
+
+<img src="../../../images/general/wpf_stackpanel.png" />
+
+If we want to achieve similar layout in Angular, we may use CSS Flexbox layout. The Flexible Box Layout Module is a powerful mechanism which allows designing flexible responsive layout structure. In order to use the Flexbox layout, we have to define a container which has its [`display`](https://www.w3schools.com/cssref/pr_class_display.asp) property set to `flex`. Also in order to stack the items vertically we have to set the [`flex-direction`](https://www.w3schools.com/cssref/css3_pr_flex-direction.asp) property to `column`.
+
+```html
+<div class="flex-container">
+    <button>1</button>
+    <button>2</button>
+    <button>3</button>
+    <button>4</button>
+    <button>5</button>
+</div>
+```
+
+```css
+.flex-container {
+    display: flex;
+    flex-flow: column;
+}
+```
+
+Here is the final result in the browser:
+
+<img src="../../../images/general/angular_stackpanel.png" />
+
+The default value of the [`flex-direction`](https://www.w3schools.com/cssref/css3_pr_flex-direction.asp) property is `row`, which is equivalent to a StackPanel with Horizontal orientation in WPF. The flexbox also supports `row-reverse` and `column-reverse` directions which stack the items right to left and bottom to top respectively.
+
+### WrapPanel
+
+The [`WarpPanel`](https://docs.microsoft.com/en-us/dotnet/api/system.windows.controls.wrappanel?view=netframework-4.8) positions child elements in sequential position from left to right, breaking content to the next line at the edge of the containing box. Subsequent ordering happens sequentially from top to bottom or right to left, depending on the value of the Orientation property. Let us add several buttons in a WrapPanel and see how they look like in WPF:
+
+```xml
+<WrapPanel>
+    <WrapPanel.Resources>
+        <Style TargetType="Button">
+            <Setter Property="Width" Value="150"></Setter>
+        </Style>
+    </WrapPanel.Resources>
+    <Button>1</Button>
+    <Button>2</Button>
+    <Button>3</Button>
+    <Button>4</Button>
+    <Button>5</Button>
+</WrapPanel>
+```
+
+<img src="../../../images/general/wpf_wrappanel.png" />
+
+In order to achieve similar result in Angular, we will use the Flexbox layout again. As in the case with StackPanel, we have to set the [`display`](https://www.w3schools.com/cssref/pr_class_display.asp) property set to `flex`, but we also have to set the [`flex-wrap`](https://www.w3schools.com/cssref/css3_pr_flex-wrap.asp) property to `wrap`.
```

Comment:
"we have to set the [`display`](https://www.w3schools.com/cssref/pr_class_display.asp) property **set** to `flex`" - remove this second "set".

Reply:
Addressed.